autoPyTorch/pipeline/tabular_regression.py

```diff
@@ -78,23 +80,37 @@ def __init__(self,
             config, steps, dataset_properties, include, exclude,
             random_state, init_params, search_space_updates)
 
-    def score(self, X: np.ndarray, y: np.ndarray, batch_size: Optional[int] = None) -> np.ndarray:
+        # Because a pipeline is passed to a worker, we need to honor the random seed
+        # in this context. A tabular regression pipeline will implement a torch
+        # model, so we comply with https://pytorch.org/docs/stable/notes/randomness.html
+        torch.manual_seed(self.random_state.get_state()[1][0])
+
+    def score(self, X: np.ndarray, y: np.ndarray,
+              batch_size: Optional[int] = None,
+              metric_name: str = 'r2') -> float:
         """Scores the fitted estimator on (X, y)
 
         Args:
-            X (np.ndarray): input to the pipeline, from which to guess targets
-            batch_size (Optional[int]): batch_size controls whether the pipeline
-                will be called on small chunks of the data. Useful when calling the
-                predict method on the whole array X results in a MemoryError.
+            X (np.ndarray):
+                input to the pipeline, from which to guess targets
+            batch_size (Optional[int]):
+                batch_size controls whether the pipeline will be
+                called on small chunks of the data. Useful when
+                calling the predict method on the whole array X
+                results in a MemoryError.
+            y (np.ndarray):
+                Ground Truth labels
+            metric_name (str, default = 'r2'):
```

Comment:
Is it from (sklearn)[https://scikit-learn.org/stable/modules/classes.html#module-sklearn.metrics], right?
Can you add it?

Reply:
no these are our names